### PR TITLE
Add support for parsing routes and PCMs in the topology parser

### PR DESCRIPTION
**Pull request by ranj063**

### CODE REVIEW

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
SOFCI TEST

**Review by btian1 — APPROVED** [review]

File: tools/tplg_parser/pcm.c [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+	pcm_info->id = pcm->pcm_id;
+
+	/* look up from the widget list and populate the PCM info */
+	list_for_item(item, widget_list) {
+		struct tplg_comp_info *comp_info = container_of(item, struct tplg_comp_info, item);
+
+		if (!strcmp(pcm->caps[0].name, comp_info->stream_name) &&
+		    (comp_info->type == SND_SOC_TPLG_DAPM_AIF_IN ||
+		     comp_info->type == SND_SOC_TPLG_DAPM_AIF_OUT)) {
```
Comment (kv2019i) [inline tools/tplg_parser/pcm.c:49]: You know this much better, but it's confusing type can be both AIF_IN and AIF_OUT for playback.

**Review by kv2019i — APPROVED** [review]
One bit I don't understand, but most probably still correct. Please @ranj063 go ahead and merge.

File: tools/tplg_parser/pcm.c (hunk 1 above)
Comment (ranj063) [reply tools/tplg_parser/pcm.c:49]: @kv2019i when we have duplex pcms, it is associated with both a playback and a capture host. So here I parse both of them

**Comment by kv2019i** [comment]
@ranj063 You can also rebase and repush, that will solve the Zephyr main failing checks in CI as well.

**Pull request merged**